Naming conventions for data package

bentrm opened the pull request:
Adds the following:
* structure stores and models under specific namespace
* use singular naming for models
* use plural naming for stores

Fixes #20

chrismayer commented on src/data/store/ObjectsCollection.js:
```diff
@@ -16,7 +16,7 @@
  * Simple store that maps a ol.Collection to a Ext.data.Store.
  */
-Ext.define('GeoExt.data.store.Collection', {
+Ext.define('GeoExt.data.store.ObjectsCollection', {
```
I know what you mean by that, but to very very strict it breaks the naming convention somehow:

> use plural naming for stores

What about calling it ``GeoExt.data.store.OlObjects``? This would bring the advantage that the name offers a hint that OpenLayers obejcts are meant to be stored in this store.

marcjansen commented:
I am fine with watever suits the most of us. If @chrismayer and @bentrm agree on a convention I agree as well.

marcjansen commented:
This will also need to be rebased.

marcjansen commented:
Wouldn't it be the best idea, to have this PR provide the base names of the classes, and to provide other names as `[alternateClassName](http://docs.sencha.com/extjs/6.0/6.0.0-classic/#!/api/Ext.Class-cfg-alternateClassName)`? We all know that naming things is very hard.

What do ypu think, @chrismayer & @bentrm?

bentrm commented:
I think it boils down to if we want to namespace stores like GeoExt 2 did with models, readers and proxies. I think it would add some structure, so I'm +1 for that. (I'm totally fine with alternate class names of which may even be added later on.) 

`GeoExt.data.store.OlObjects` may be a better suited class name.This PR should probably also rename `GeoExt.data.model.Object` to `OlObject` to prevent ambiguity?

chrismayer commented:
Yes @bentrm, I also prefer ``GeoExt.data.store.OlObjects`` because of the speaking name. As you wrote it would be consequent to rename ``GeoExt.data.model.Object`` to ``OlObject`` as well.

@marcjansen I am of course fine to add ``alternateClassName``, where it makes sense. I leave this decision whether to add ``alternateClassName`` or not in this case up to @bentrm.

marcjansen commented:
Any updates on this PR, @bentrm and @chrismayer?

I want pur classnames to make sense and be consistent. Yet I want backwatds compatibility.

marcjansen commented:
Hey @bentrm and @chrismayer, is anybody still interested in getting this in? AFAICT only a decision needs to be made, right?

chrismayer commented:
Yes, but I think it the descision is not that problematic. But I guess in the meantime we ran into some conflicts which have to be resolved.

To make it a bit quicker I propose to rename ```GeoExt.data.store.Collection``` in ```GeoExt.data.store.OlObjects```  and ```GeoExt.data.model.Object``` to ```GeoExt.data.model.OlObject```.

@bentrm do you have time to apply this?

bentrm commented:
@marcjansen, @chrismayer, just added your proposal as we pretty much agreed on that before anyway (OlObject, OlObjects).
Also, I made the following changes:
* renamed GeoExt.model.print.LayoutAttributes to LayoutAttribute (which makes sense to me, as the entity describes a single attribute of the layout which happens to have attributes of name and type itself)
* renamed GeoExt.store.Layer to Layers
* renamed GeoExt.store.Tree to LayersTree (Which to me is more descriptive. Of course, this is kind of out of line with what I had proposed in the beginning.)

I was wondering to even nest Tree into GeoExt.store.tree.Layers as one may add other tree stores later on? Maybe stuff like a geometry tree that allows to edit nested geometries as in GeoJSON (just thinking out loud here, please don't :goberserk:). 
What do you think?

marcjansen commented:
As I already stated. If @bentrm and @chrismayer agree on sth. I am :+1:

/cc @weskamm and @kaivolland. A possible merge may influence your projects

weskamm commented:
Please also rename the test files according to your naming changes.
Otherwise it looks good to me

chrismayer commented:
Thanks @bentrm for your ongoing effort. This looks good to me (besides the point @weskamm stated with the adaption of the file names).

bentrm commented:
> Please also rename the test files according to your naming changes.
:heavy_check_mark:

marcjansen commented:
@chrismayer please merge this if you are happy with this.

marcjansen commented:
Tjanks for the continued work on this!

chrismayer commented:
Thank you all for your work on this!
